.github/workflows/inspektor-gadget.yml

```diff
@@ -195,25 +210,103 @@ jobs:
           ENABLE_BTFGEN=true
-        # At the moment, we only build core image and do not publish them.
-        push: ${{ matrix.type != 'core' }}
+        outputs: type=docker,dest=/tmp/gadget-container-image-${{ matrix.type }}-${{ matrix.os }}-${{ matrix.platform }}.tar
         tags: ${{ steps.choose-repo-determine-image-tag.outputs.container-repo }}:${{ steps.choose-repo-determine-image-tag.outputs.image-tag }}
         cache-from: type=local,src=/tmp/.buildx-cache
-        cache-to: type=local,dest=/tmp/.buildx-cache-new
-        # For the moment, we only support these two platforms.
-        platforms: linux/amd64,linux/arm64
+        cache-to: type=local,dest=/tmp/.buildx-cache-docker
+        platforms: ${{ matrix.os }}/${{ matrix.platform }}
+
+    - name: Publish gadget ${{ matrix.type }} ${{ matrix.os }} ${{ matrix.platform }} container image as artifacts
+      uses: actions/upload-artifact@master
+      with:
+        name: gadget-container-image-${{ matrix.type }}-${{ matrix.os }}-${{ matrix.platform }}.tar
+        path: /tmp/gadget-container-image-${{ matrix.type }}-${{ matrix.os }}-${{ matrix.platform }}.tar
+        retention-days: 1
+
+    # build time will not be increased with this workflow because of internal cache
+    # buildx is used here since it allows push-by-digest to avoid platform specific tags
+    - name: Publish gadget ${{ matrix.type }} ${{ matrix.os }} ${{ matrix.platform }} container image to registry
+      id: publish-gadget-container-images
+      if: github.event_name != 'pull_request'
+      uses: docker/build-push-action@v2
+      with:
+        context: /home/runner/work/inspektor-gadget/inspektor-gadget/
+        file: /home/runner/work/inspektor-gadget/inspektor-gadget/Dockerfiles/gadget-${{ matrix.type }}.Dockerfile
+        build-args: |
+          ENABLE_BTFGEN=true
+        outputs: type=registry,push=true,push-by-digest=true
+        tags: ${{ steps.choose-repo-determine-image-tag.outputs.container-repo }}
+        cache-from: type=local,src=/tmp/.buildx-cache-docker
+        cache-to: type=local,dest=/tmp/.buildx-cache-registry
+        platforms: ${{ matrix.os }}/${{ matrix.platform }}
+
+    - name: Save gadget ${{ matrix.type }} ${{ matrix.os }} ${{ matrix.platform }} container image digest output
+      id: published-gadget-container-images
+      if: github.event_name != 'pull_request'
+      run: |
+          echo "::set-output name=${{ matrix.type }}-${{ matrix.platform }}::${{ steps.publish-gadget-container-images.outputs.digest }}"
+
+  publish-gadget-images-manifest:
+    name: Publish gadget container images manifest
+    # level: 1
+    if: github.event_name != 'pull_request'
+    needs: build-gadget-container-images
+    runs-on: ubuntu-latest
+    permissions:
+      # allow publishing container image
+      # in case of public fork repo/packages permissions will always be read
+      contents: read
+      packages: write
+    strategy:
+      matrix:
+        type: [ default ]
```

Comment:
@eiffel-fl just wanted to get your thoughts here for a UX thing. GH package only shows [single tag](https://github.com/kinvolk/inspektor-gadget/pkgs/container/inspektor-gadget) on the package page meaning that user will be either see `latest` or `latest-core` depending on which matrix run wins. So maybe once we enable core image creation we get rid of the matrix and do it in a sequence for consistency. What do you think?

Reply:
I meant in the header like above this list you posted as below:

![image](https://user-images.githubusercontent.com/10230890/188410898-84dc42e2-f4d4-4d8c-8d33-d4c45cd910da.png)

Reply:
Okay Perfect.

Reply:
@mqasimsarfraz you're right. The last one to be run there will be shown on the header... (AFAIU GitHub doesn't treat `lastest` in a special way, the latest pushed tag is shown there). Anyway, I don't think it's a bit issue since we don't expect users to pull containers by hand.